src/api/FactoryAPI.ts

```diff
@@ -138,6 +138,78 @@ export class FactoryAPI {
     return new BigNumber(10 ** (18 - minDecimal.toNumber()));
   }
```

Comment:
As a reference, look at this [spreadsheet](https://docs.google.com/spreadsheets/d/1aiwkynYYjcRqJUKvrV0UzCA0H5xlU_pIzINuauxeBW8/edit?usp=sharing) around Line 56 on how to calculate 1) the required component units, 2) ideal natural unit, and 3) finalized component units. This can be written in 3 sub-functions.

Step 1
```
function calculateRequiredComponentUnits(
  components: Address[],
  componentPrices: BigNumber[],
  desiredPercentages: BigNumber[],
  targetPrice: BigNumber
): BigNumber[] {
  // Get component decimals
  // Calculate the required amount of components to match the desired percentages
}
```
Step 2
```
function calculateIdealNaturalUnit(
  requiredComponents: BigNumber[] // From Step 1
): BigNumber {
  // Calculate the min natural unit
}
```

Step 3
```
calculateFinalUnits(
  requiredComponentUnits: BigNumber[], // From above
  naturalUnit: BigNumber // Calculated above
): BigNumber[] {
  // Perform rounding calculations
}
```

Reply:
i already have the calculating of the minimum natural unit. do we always want the minimum though? I thought we might want to change it sometimes depending on the price and percentages we want?

Reply:
i think this is basically what i have though just with different naming

Reply:
I think we need the natural unit, because it is also an input into the `create` function